src/main/java/com/uid2/admin/vertx/service/SharingService.java

```diff
@@ -0,0 +1,187 @@
+package com.uid2.admin.vertx.service;
+
+import com.uid2.admin.store.reader.ISiteStore;
+import com.uid2.admin.store.writer.KeyAclStoreWriter;
+import com.uid2.admin.vertx.WriteLock;
+import com.uid2.shared.auth.EncryptionKeyAcl;
+import com.uid2.shared.auth.Role;
+import com.uid2.shared.middleware.AuthMiddleware;
+import com.uid2.shared.store.reader.RotatingKeyAclProvider;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class SharingService implements IService {
+    private final AuthMiddleware auth;
+
+    private final WriteLock writeLock;
+    private final KeyAclStoreWriter storeWriter;
+    private final RotatingKeyAclProvider keyAclProvider;
+    private final ISiteStore siteProvider;
+    private static final Logger LOGGER = LoggerFactory.getLogger(SharingService.class);
+
+    public SharingService(AuthMiddleware auth,
+                          WriteLock writeLock,
+                          KeyAclStoreWriter storeWriter,
+                          RotatingKeyAclProvider keyAclProvider,
+                          ISiteStore siteProvider) {
+        this.auth = auth;
+        this.writeLock = writeLock;
+        this.storeWriter = storeWriter;
+        this.keyAclProvider = keyAclProvider;
+        this.siteProvider = siteProvider;
+    }
+
+    @Override
+    public void setupRoutes(Router router) {
+        router.get("/api/sharing/list/get").handler(
```

Comment:
Are we using the term 'list' for anything? In an API URL I'd normally expect names and plurals - so maybe:
/api/sharing/sites
/api/sharing/sites/:siteid
That kind of thing.

Reply:
I was thinking lists because we are getting/setting lists not sites

Reply:
Yeah, let's run with it then :)